### DISCUSSION TITLE
Using global options

### DISCUSSION THREAD

**Discussion by mattansb** [easystats/easystats#209] [Ideas]
We don't use `options()` or `getOption()` in {easystats} - is there a reason? 

I was thinking of implementing these for control of CIs (for the `ci` and `alternative` arguments), something along the line of:

```r
# in zzz.R
.onLoad <- function(libname, pkgname) {
  op <- options()
  op.es <- list(
    es.ci.mag = 0.95,
    es.ci.diff = 0.95,
    es.alt.mag = "greater",
    es.alt.diff = "two.sided"
  )

  toset <- !names(op.es) %in% names(op)
  if(any(toset)) options(op.afex[toset])

  invisible(NULL)
}
```

Then functions would be defined:

```r
eta_squared <- function(..., 
                        ci = getOption("es.ci.mag", 0.95), 
                        alternative = getOption("es.alt.mag", "greater"),
                        ...) {
  # ...
}


cohens_d <- function(..., 
                     ci = getOption("es.ci.diff", 0.95), 
                     alternative = getOption("es.alt.diff", "two.sided"),
                     ...) {
  # ...
}
```

This could also be useful in `bayestestR` (where I personally am a fan of the 89% CI... 🙃)

**Comment by strengejacke** [comment]
Yeah, I was thinking about this, too, e.g. I often use `model_parameters(summary = TRUE)`, so customizing settings would be nice. It requires a good documentation, so it's accessible or visible to users. Maybe also in our meta-package?

**Reply by mattansb** [reply]
We can have them where ever we need them!

My only concern is that users might find it opaque :/ So good clear documentation is indeed required...

**Comment by strengejacke** [comment]
![image](https://user-images.githubusercontent.com/26301769/152639203-b6bb8042-2611-47d1-bb28-05f92f3987ab.png)

**Comment by strengejacke** [comment]
ôr

![image](https://user-images.githubusercontent.com/26301769/152639217-3131a018-fb68-4f39-ba09-8765b7c707e6.png)

**Comment by strengejacke** [comment]
The question is: do we want `getOption()` in the function definition (like in your examples), which might be confusing to new users and really looks a bit ugly...

Or do we want to check inside a function, like:

```r
foo <- function(ci = .95) {
  if (missing(ci)) {
    ci <- getOption("ci", .95)
  }
}
```

**Comment by bwiernik** [comment]
I think `ci = getOption("easystats.ci", defaultValue = .95)` probably isn't too alarming or confusing. I would say we should have just a single global CI option.
